src/config.rs

```diff
@@ -196,7 +196,12 @@ impl Cfg {
 
         // Centralised file for multi-user systems to provide admin/distributor set initial values.
         let fallback_settings = if cfg!(not(windows)) {
-            FallbackSettings::new(PathBuf::from(UNIX_FALLBACK_SETTINGS))?
+            // If present, use the RUSTUP_OVERRIDE_UNIX_FALLBACK_SETTINGS environment
+            // variable as settings path, or UNIX_FALLBACK_SETTINGS otherwise
+            FallbackSettings::new(PathBuf::from(
+                option_env!("RUSTUP_OVERRIDE_UNIX_FALLBACK_SETTINGS")
+                    .unwrap_or(UNIX_FALLBACK_SETTINGS),
```

Comment:
`option_env!()` is a compile-time expansion of the environment variable.

You probably meant to retrieve the runtime environment which we do via our process code as:

```rust
process().var("RUSTUP_OVERRIDE_UNIX_FALLBACK_SETTINGS").unwrap_or(UNIX_FALLBACK_SETTINGS)
```